docs/CaliptraHardwareSpecification.md

```diff
@@ -92,15 +107,14 @@ The following table shows the memory map address ranges for each of the IP block
 | Data Vault                          | 5         | 8 KiB        | 0x1001_C000   | 0x1001_DFFF |
 | SHA512                              | 6         | 32 KiB       | 0x1002_0000   | 0x1002_7FFF |
 | SHA256                              | 13        | 32 KiB       | 0x1002_8000   | 0x1002_FFFF |
+| ML-DSA                              | 17        | 64 KiB       | 0x1003_0000   | 0x1003_FFFF |
 
 #### Peripherals subsystem
```

Comment:
Let's also update the interrupt vector list here: 
https://github.com/chipsalliance/caliptra-rtl/blob/6cd75c8b6d71e87ef9a0d1ba8bc0596651183770/docs/CaliptraHardwareSpecification.md#external-interrupts

The latest and greatest list of interrupt vectors is in [this file](https://github.com/chipsalliance/caliptra-rtl/pull/600/files#diff-052012e084cb1592561733e71302a9cbfebf3d3cfe4a422856af1049f13fad33)